tests/test_mailchimp_bookmarks.py

```diff
@@ -0,0 +1,240 @@
+                        "list_members": {"datetime": "2022-08-01T06:23:35+00:00"}
+                    }
+                },
+                "reports_email_activity_next_chunk": 0,
+                "reports_email_activity_last_run_id": None,
+                "reports_email_activity": {  # Verifying bookmark for single campaign's 'reports_email_activity'
+                    "32e6edcecb": "2016-05-15T18:57:16+00:00"
+                },
+            }
+        }
+
+        menagerie.set_state(conn_id, new_states)
+
+        ##########################################################################
+        # Second Sync
+        ##########################################################################
+
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Test By Stream
+        ##########################################################################
+
+        bookmark_path = self.get_bookmark_path()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # Expected values
+                expected_replication_method = expected_replication_methods[stream]
+                stream_bookmark_path = bookmark_path.get(stream)
+
+                # Collect information for assertions from syncs 1 & 2 based on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [
+                    record.get("data")
+                    for record in first_sync_records.get(stream, {}).get("messages", [])
+                    if record.get("action") == "upsert"
+                ]
+                second_sync_messages = [
+                    record.get("data")
+                    for record in second_sync_records.get(stream, {}).get(
+                        "messages", []
+                    )
+                    if record.get("action") == "upsert"
+                ]
+                first_bookmark_value = self.get_bookmark(
+                    first_sync_bookmarks.get("bookmarks"), stream_bookmark_path
+                )
+                second_bookmark_value = self.get_bookmark(
+                    second_sync_bookmarks.get("bookmarks"), stream_bookmark_path
+                )
+
+                if expected_replication_method == self.INCREMENTAL:
+                    # Get parent key in child's record
+                    parent_id = (
+                        "campaign_id"
+                        if stream == "reports_email_activity"
+                        else "list_id"
+                    )
+                    # Get parent id from bookmark
+                    parent_id_value = stream_bookmark_path[1]
```

Comment:
Are we validating bookmarks for only one `parent_id`? Why we aren't iterating through other parent_ids as well?